### DISCUSSION TITLE
How can i use multiples gateways in checkout page?

### DISCUSSION THREAD

**Discussion by heycesar** [duncanmcclean/simple-commerce#796] [Help] [answered]
I have a checkout page where I would to use stripe and paypal payments, but i don't find a way to merge both methods in a single flow to pay the products, there is a way to build this feature?

In this moment just work with two `{{ sc:checkout }}` sections, but I would like to use some fields for order register.

**Comment by duncanmcclean** [comment, marked as answer]
Hey!

Sure, you can loop through each of the gateways (Stripe & PayPal in your case), then let the user select which one they want to use and show the right form (with Vue/Alpine/whatever you're using)

In fact, a great example of this happening is in the Simple Commerce [Starter Kit](https://github.com/duncanmcclean/sc-starter-kit/blob/main/resources/views/checkout/payment.antlers.html).

Hopefully that example should help!

**Reply by heycesar** [reply]
Oh great, thanks @duncanmcclean, and additional question, sorry for bother you,

I have a weird behavior with Paypal, I'm using onsite mode, I already configure the correct webhook in sandbox mode, looks like working when i pay, even redirect to thank you page, create the order with de data in de dashboard as well, but the order always generated as no paid.

I hope can you help me.

**Reply by duncanmcclean** [reply]
Hmm, do you have something like Ngrok/Expose setup for the webook?

You'll need something like that so PayPal can talk to your local development server.

**Reply by heycesar** [reply]
Yeah, actually i'm using expose to setup de public url in paypal webhook

**Reply by duncanmcclean** [reply]
Okay, a couple of other questions:

1. If you look at Expose, it should tell you the requests being made, do you see the webhook requests happen after you go through the PayPal process?
2. If you do, does Expose say those requests returned 500 errors? If so, you should be able to see a PayPal/gateway related error in your log file (`storage/logs`), would you be able to post what it is?
3. Where are you checking the payment status - in the Control Panel or are you checking `is_paid` directly in the order entry's file?

**Reply by heycesar** [reply]
Hi @duncanmcclean I was the error LOL, I was exposing bad the url project.

Thank a lot for your help.

Have a nice day!

**Reply by duncanmcclean** [reply]
No worries! I had a feeling that was the case (happens a lot with off-site gateways 😆 )

You too!
